modules/apps/cookies/cookies-banner-web/src/main/resources/META-INF/resources/cookies_banner/js/CookiesBanner.js

```diff
@@ -187,4 +187,33 @@ function setBannerVisibility(cookieBanner) {
 	}
 }
 
-export {openCookieConfigurationModal};
+function checkCookieConsentForTypes(cookieTypes, modalOptions) {
+	function areAccepted(cookieTypes) {
+		if (Array.isArray(cookieTypes)) {
+			return cookieTypes.reduce(
+				(acc, cur) => acc && checkConsent(cur),
+				true
+			);
+		}
+		else {
+			return checkConsent(cookieTypes);
+		}
+	}
+
+	return new Promise((resolve, reject) => {
+		if (areAccepted(cookieTypes)) {
+			resolve();
```

Comment:
@bryceosterhaus @kresimir-coko I've added this thing as a new API so that dxp apps don't need to implement complex logic to handle the cookie banner interaction. 

In this case I'm using promises to wait for the "right" user input. I could not find much cases like this in the JS apis we provide, so I'd like to hear your thoughts